docs/TUTORIAL.md

```diff
@@ -1,3 +1,413 @@
+
+        container.portsType.containerPort(conf.ingressController.container.ports.https) +
+        container.portsType.hostPort(conf.ingressController.container.ports.https),
+    ]);
+
+local ingressController = deployment.new(conf.ingressController.name, conf.ingressController.replicas, [nginxContainer], conf.ingressController.selector) +
+    deployment.mixin.metadata.mixinInstance(controllerMetadata) +
+    deployment.mixin.spec.template.spec.terminationGracePeriodSeconds(conf.ingressController.terminationGracePeriodSeconds);
+
+// add the new Service object
+local ingressService = service.new(conf.ingressController.name, conf.ingressController.selector, [
+    service.mixin.specType.portsType.name("http") +
+    service.mixin.specType.portsType.port(conf.ingressController.container.ports.http),
+
+    service.mixin.specType.portsType.name("https") +
+    service.mixin.specType.portsType.port(conf.ingressController.container.ports.https)
+]) +
+    service.mixin.metadata.mixinInstance(controllerMetadata) +
+    service.mixin.spec.type("LoadBalancer");
+
+// call the modified object that now includes the deployment and the service
+k.core.v1.list.new([ingressController, ingressService])
+```
+
+Save the file again, and run again:
+
+```bash
+jsonnet ingressController.jsonnet > deployment.yaml
+kubectl apply -f deployment.yaml
+```
+
+The generated YAML now looks like this:
+
+```yaml
+apiVersion: v1
+items:
+  - apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      labels:
+        run: nginx-ingress-controller
+      name: nginx-ingress-controller
+      namespace: nginx-ingress
+    spec:
+      replicas: 2
+      template:
+        metadata:
+          labels:
+            run: nginx-ingress-controller
+        spec:
+          containers:
+            - image: 'gcr.io/google_containers/nginx-ingress-controller:0.9.0-beta.7'
+              name: nginx-ingress-controller
+              ports:
+                - containerPort: 80
+                  hostPort: 80
+                - containerPort: 443
+                  hostPort: 443
+          terminationGracePeriodSeconds: 60
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      labels:
+        run: nginx-ingress-controller
+      name: nginx-ingress-controller
+      namespace: nginx-ingress
+    spec:
+      ports:
+        - name: http
+          port: 80
+        - name: https
+          port: 443
+      selector:
+        run: nginx-ingress-controller
+      type: LoadBalancer
+kind: List
+```
+
+You can also easily add the built-in mixins for liveness and readiness probes to the container object.
+
+## Write a custom mixin to include as a sidecar
+
+Your application needs logging and monitoring, so you can write a mixin to define a 
+Fluentd container. You can then add it to your cluster as a sidecar.
+
+Because mixins let you assemble components at runtime (dynamic inheritance),
```

Comment:
"Dynamic inheritance" might be a little confusing to people who are not in the know. It might be better to talk about it like a fancy object merge.